actor/actor_system.go

```diff
@@ -2921,16 +2921,9 @@ func (x *actorSystem) checkSpawnPreconditions(ctx context.Context, actorName str
 		}
 
 		// here we make sure in cluster mode that the given actor is uniquely created
-		// by checking both its kind and identifier
-		existed, err := x.cluster.GetActor(ctx, actorName)
-		if err != nil {
-			if errors.Is(err, cluster.ErrActorNotFound) {
-				return nil
-			}
+		if exists, err := x.cluster.ActorExists(ctx, actorName); err != nil {
```

Comment:
```suggestion
		exists, err := x.cluster.ActorExists(ctx, actorName)
		
	  if err != nil {
		  return err
	  } 
	  
	  if exists {
		  return NewErrActorAlreadyExists(actorName)
	  }
		
```

I try as much as possible avoid nested if conditions